src/scaffoldmaker/meshtypes/meshtype_3d_bladderurethra1.py

```diff
@@ -0,0 +1,879 @@
+        urethraDiameter2 = options['Urethra diameter 2']
+        urethraWallThickness = options['Urethra wall thickness']
+        urethraproportion = options['Urethra length proportion to central path']
+
+        centralPath = options['Central path LUT']
+
+        includeUreter = options['Include ureter']
+        ostiumOptions = options['Ureter']
+        ostiumDefaultOptions = ostiumOptions.getScaffoldSettings()
+        elementsCountAroundOstium = ostiumDefaultOptions['Number of elements around ostium']
+        elementsCountAnnulusRadially = options['Number of elements radially on annulus']
+        ostiumPositionAround = options['Ostium position around']
+        ostiumPositionDown = options['Ostium position down']
+
+        firstNodeIdentifier = 1
+        firstElementIdentifier = 1
+
+        fm = region.getFieldmodule()
+        fm.beginChange()
+        nodes = fm.findNodesetByFieldDomainType(Field.DOMAIN_TYPE_NODES)
+        mesh = fm.findMeshByDimension(3)
+
+        # Central path
+        tmpRegion = region.createRegion()
+        centralPath.generate(tmpRegion)
+        cx, cd1, cd2, cd12 = extractPathParametersFromRegion(tmpRegion)
+        # for i in range(len(cx)):
+        #     print(i, '[', cx[i], ',', cd1[i], ',', cd2[i], ',', cd12[i], '],')
+        del tmpRegion
+
+        # Find arcLength
+        length = 0.0
+        elementsCountIn = len(cx) - 1
+        sd1 = interp.smoothCubicHermiteDerivativesLine(cx, cd1, fixAllDirections=True,
+                                                       magnitudeScalingMode=interp.DerivativeScalingMode.HARMONIC_MEAN)
+        for e in range(elementsCountIn):
+            arcLength = interp.getCubicHermiteArcLength(cx[e], sd1[e], cx[e + 1], sd1[e + 1])
+            length += arcLength
+        segmentLength = length / elementsCountAlong
+
+        if includeUrethra:
+            urethraLength = urethraproportion * length
+            elementsCountAlongUrethra = int(urethraLength / segmentLength)
+            elementsCountAlongBladder = elementsCountAlong - elementsCountAlongUrethra
+        else:
+            elementsCountAlongBladder = elementsCountAlong
+
+        # Sample central path
+        # sx, sd1, se, sxi, ssf = interp.sampleCubicHermiteCurves(cx, cd1, elementsCountAlong)
+        # sd2, sd12 = interp.interpolateSampleCubicHermite(cd2, cd12, se, sxi, ssf)
+        if includeUrethra:
+            sx, sd1, se, sxi, ssf = interp.sampleCubicHermiteCurves(cx, cd1, elementsCountAlong)
+            sd2, sd12 = interp.interpolateSampleCubicHermite(cd2, cd12, se, sxi, ssf)
+        else:
+            m = int((1 - urethraproportion) * len(cx))
+            cx = cx[:m+1]
+            cd1 = cd1[:m+1]
+            cd2 = cd2[:m+1]
+            cd12 = cd12[:m+1]
+            sx, sd1, se, sxi, ssf = interp.sampleCubicHermiteCurves(cx, cd1, elementsCountAlongBladder)
+            sd2, sd12 = interp.interpolateSampleCubicHermite(cd2, cd12, se, sxi, ssf)
```

Comment:
Not needed with annotated central path.

Reply:
If I delete this part, how I can run the following part?

        # Project reference point for warping onto central path
        sxRefList, sd1RefList, sd2ProjectedListRef, zRefList = \
             tubemesh.getPlaneProjectionOnCentralPath(innerNodes_x, elementsCountAround, elementsCountAlong,
                                                      length, sx, sd1, sd2, sd12)

Reply:
You can still do this on each respective sets of cx, cd1 for the bladder and urethra.